File: pet-service/src/main/java/com/phorus/petservice/services/PetService.java
package com.phorus.petservice.services;

import com.phorus.petservice.model.dbentities.Pet;
import com.phorus.petservice.model.dtos.PetDTO;


public interface PetService {

    Pet findById(Long petId);

    Long create(Pet pet);

    void update(Long petId, PetDTO petDTO);

    void deleteById(Long petId);


    Long getAmountOfPets(Long userId);
}

File: pet-service/src/main/kotlin/com/phorus/petservice/services/PetService.kt
package com.phorus.petservice.services

import com.phorus.petservice.model.dbentities.Pet
import com.phorus.petservice.model.dtos.PetDTO

interface PetService {

    suspend fun findById(petId: Long): Pet

    suspend fun create(pet: Pet): Long

    suspend fun update(petId: Long, petDTO: PetDTO)

    suspend fun deleteById(petId: Long)

    suspend fun getAmountOfPets(userId: Long): Long
}